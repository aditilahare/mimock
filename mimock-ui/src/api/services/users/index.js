import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';

export { getAllUsers, deleteUser };
