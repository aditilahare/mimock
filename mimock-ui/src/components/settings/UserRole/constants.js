export const constants = {
	testIds: {
		userRoleContainer: 'user-role-container',
		userRoleHeader: 'user-role-header',
		roleInfoMessage: 'role-info-message',
	},
	label: {
		roleHeader: 'Current Role',
		roleMessage: 'Reach out to your ADMIN to update your role',
	},
};
