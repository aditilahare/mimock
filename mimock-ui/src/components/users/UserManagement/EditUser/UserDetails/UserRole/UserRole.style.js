import tw from 'tailwind-styled-components';
import { BsQuestionCircle } from 'react-icons/bs';
import { CustomButton } from 'styles';
import { Info, Label } from '../UserDetails.style';

export const UserRoleWrapper = tw(Info)``;

export const UserRoleLabel = tw(Label)``;

export const UserRoleActions = tw.div`
    w-1/2
    flex
    items-center
    align-middle
    ${(props) => (props.$margin ? props.$margin : 'mx-20')}
`;

export const RoleOptions = tw.select`
    font-sans
    rounded-md
    shadow-md
    font-semibold
    p-2
    border-2
    border-gray-200
    outline-none
    mr-2
`;

export const RoleHint = tw(BsQuestionCircle)`
    text-gray-500
    text-2xl
`;

export const UpdateRoleButton = tw(CustomButton)``;
