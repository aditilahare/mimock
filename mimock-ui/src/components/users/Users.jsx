import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
