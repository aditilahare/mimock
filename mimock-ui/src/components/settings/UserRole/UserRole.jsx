import React from 'react';
import { PropTypes } from 'prop-types';
import RolePill from 'users/UsersTable/UserRow/RolePill';
import { constants } from './constants';
import {
	UserRoleContainer,
	RoleContent,
	UserRoleHeader,
	InfoMessageContainer,
	Message,
	InfoIcon,
} from './UserRole.style';

function UserRole({ role }) {
	// #region Defaults
	const { label, testIds } = constants;
	// #endregion

	return (
		<UserRoleContainer data-testid={testIds.userRoleContainer}>
			<UserRoleHeader data-testid={testIds.userRoleHeader}>
				{label.roleHeader}
			</UserRoleHeader>
			<RoleContent>
				<RolePill role={role} margin='mx-0' />
				<If condition={role !== 'ADMIN'}>
					<InfoMessageContainer
						data-testid={testIds.roleInfoMessage}
						role={role}
					>
						<InfoIcon />
						<Message>{label.roleMessage}</Message>
					</InfoMessageContainer>
				</If>
			</RoleContent>
		</UserRoleContainer>
	);
}

UserRole.propTypes = {
	role: PropTypes.string.isRequired,
};

export default UserRole;
