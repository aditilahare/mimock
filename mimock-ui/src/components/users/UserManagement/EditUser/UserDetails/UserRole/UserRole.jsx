import React, { useState, useEffect } from 'react';
import useNotification from 'hooks/useNotification';
import { getUserRoles, updateUserRole } from 'services/users';
import { ConfirmationModal } from 'components/common/Modals';
import { useRecoilState } from 'recoil';
import editUserDetailsAtom from 'atoms/editUserDetailsAtom';
import { notificationTypes } from 'constants/notificationConstants';
import UserRoleOptions from './UserRoleOptions';
import { UserRoleWrapper, UserRoleLabel } from './UserRole.style.js';

function UserRole() {
	const [userInfo, setUserInfo] = useRecoilState(editUserDetailsAtom);
	const { userName, userRole: currentUserRole } = userInfo;

	const [roles, setRoles] = useState([]);
	const [selectedRole, setSelectedRole] = useState(currentUserRole);
	const [selectedRoleDescription, setSelectedRoleDescription] = useState('');
	const [roleFetchError, setRoleFetchError] = useState(false);
	const [updatingUserRole, setUpdatingUserRole] = useState(false);
	const [showUpdateConfirmationModal, setShowUpdateConfirmationModal] =
		useState(false);

	const updateConfirmationMessage = `Are you sure you want to update user role to "${selectedRole}" ?`;
	const updatingMessage = 'Updating user role. Please wait...';

	useEffect(() => {
		setSelectedRole(currentUserRole);

		getUserRoles()
			.then((res) => {
				setRoles(res);
				setSelectedRoleDescription(
					res.find((role) => role.roleName === currentUserRole)?.roleDescription
				);
			})
			.catch(() => {
				setRoleFetchError(true);
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_ERROR,
					title: `Failed to fetch role for user`,
					message: 'Excluding role modification from edit user form',
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
			});
	}, [currentUserRole, useNotification]);

	const updateRole = async () => {
		setUpdatingUserRole(true);
		updateUserRole(userName, selectedRole)
			.then(() => {
				setUpdatingUserRole(false);
				setShowUpdateConfirmationModal(false);
				setUserInfo({
					...userInfo,
					userRole: selectedRole,
				});
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_SUCCESS,
					title: 'User role updated successfully',
					message: `User role updated to ${selectedRole}`,
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
			})
			.catch(() => {
				setUpdatingUserRole(false);
				setShowUpdateConfirmationModal(false);
				useNotification({
					type: notificationTypes.NOTIFICATION_TYPE_ERROR,
					title: 'Failed to update user role',
					message: 'Please try again',
					animationIn: 'animate__slideInRight',
					animationOut: 'animate__slideOutRight',
				});
			});
	};

	return (
		<If condition={!roleFetchError}>
			<If condition={showUpdateConfirmationModal}>
				<ConfirmationModal
					message={updateConfirmationMessage}
					confirmButtonLabel='Update user role'
					loading={updatingUserRole}
					loadingMessage={updatingMessage}
					onConfirm={async () => {
						await updateRole();
					}}
					onCancel={() => {
						setShowUpdateConfirmationModal(false);
					}}
				/>
			</If>
			<UserRoleWrapper data-testid='edit-user-role'>
				<UserRoleLabel>Role</UserRoleLabel>
				<UserRoleOptions
					roles={roles}
					selectedRole={selectedRole}
					setSelectedRole={setSelectedRole}
					enableRoleUpdate={true}
					currentUserRole={currentUserRole}
					selectedRoleDescription={selectedRoleDescription}
					setSelectedRoleDescription={setSelectedRoleDescription}
					setShowUpdateConfirmationModal={setShowUpdateConfirmationModal}
				/>
			</UserRoleWrapper>
		</If>
	);
}

export default UserRole;
