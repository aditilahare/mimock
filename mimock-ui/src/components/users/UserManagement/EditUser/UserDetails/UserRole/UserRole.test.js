import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import { useRecoilState } from 'recoil';
import { getUserRoles, updateUserRole } from 'services/users';
import UserRole from './UserRole';

const actualNotification = jest.requireActual('hooks/useNotification');

let mockedRecoilFn = jest.fn();
jest.mock('recoil');
jest.mock('services/users');
jest.mock('react-notifications-component', () => {
	const Store = {
		addNotification: jest.fn(),
	};
	return { Store };
});

describe('UserRole', () => {
	beforeEach(() => {
		useRecoilState.mockImplementation(() => {
			return [
				{
					userName: 'test1',
					name: 'Test User',
					isUserActive: false,
					userRole: 'MANAGER',
					userCreatedAt: '2022-04-30 20:35:35',
					passwordUpdatedAt: '2022-04-30 20:35:35',
				},
				mockedRecoilFn,
			];
		});
		getUserRoles.mockResolvedValue([
			{
				roleName: 'ADMIN',
				roleDescription:
					'Administrator can create and manage mocks, and also has user management privileges',
			},
			{
				roleName: 'MANAGER',
				roleDescription:
					'Manager can create and manage mocks, but cannot manage users',
			},
			{
				roleName: 'VIEWER',
				roleDescription: 'Viewer can only view mocks',
			},
		]);
	});

	it('should render user role', async () => {
		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId, container, rerender } = tree;

		await rerender(<UserRole />);

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(queryByTestId('update-role-btn')).not.toBeInTheDocument();

		expect(getByTestId('user-role-options')).toBeInTheDocument();
		expect(getByTestId('user-role-options')).toHaveTextContent('MANAGER');
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getByTestId('user-role-tooltip')).toBeInTheDocument();

		expect(getUserRoles).toHaveBeenCalledTimes(1);

		expect(container).toMatchSnapshot();
	});

	it('should show error banner when get user role call fails', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		getUserRoles.mockRejectedValue(new Error('Error'));

		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { queryByTestId, rerender } = tree;

		await rerender(<UserRole currentUserRole='MANAGER' />);

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(queryByTestId('update-role-btn')).not.toBeInTheDocument();
		expect(queryByTestId('edit-user-role')).not.toBeInTheDocument();

		expect(getUserRoles).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'Excluding role modification from edit user form',
			title: 'Failed to fetch role for user',
			type: 'danger',
		});
	});

	it('should show update role button on changing user role', async () => {
		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId, container } = tree;

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getUserRoles).toHaveBeenCalledTimes(1);

		await act(async () => {
			fireEvent.change(getByTestId('user-role-options'), {
				target: { value: 'ADMIN' },
			});
		});

		expect(getByTestId('update-role-btn')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should show update role confirmation modal on clicking update role button', async () => {
		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId } = tree;

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getUserRoles).toHaveBeenCalledTimes(1);

		await act(async () => {
			fireEvent.change(getByTestId('user-role-options'), {
				target: { value: 'ADMIN' },
			});
		});

		await act(async () => {
			fireEvent.click(getByTestId('update-role-btn'));
		});

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});

	it('should close update role confirmation modal on clicking cancel', async () => {
		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId } = tree;

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getUserRoles).toHaveBeenCalledTimes(1);

		await act(async () => {
			fireEvent.change(getByTestId('user-role-options'), {
				target: { value: 'ADMIN' },
			});
			fireEvent.click(getByTestId('update-role-btn'));
		});
		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('confirmation-modal-cancel-btn'));
		});
		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});

	it('should update role on clicking confirm button', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updateUserRole.mockResolvedValue({
			userName: 'frodo1',
			updatedUserRole: 'MANAGER',
		});

		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId } = tree;

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getUserRoles).toHaveBeenCalledTimes(1);

		await act(async () => {
			fireEvent.change(getByTestId('user-role-options'), {
				target: { value: 'ADMIN' },
			});
			fireEvent.click(getByTestId('update-role-btn'));
		});
		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('confirmation-modal-confirm-btn'));
		});
		expect(updateUserRole).toHaveBeenCalledTimes(1);
		expect(updateUserRole).toHaveBeenCalledWith('test1', 'ADMIN');
		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(mockedRecoilFn).toHaveBeenCalledWith({
			isUserActive: false,
			name: 'Test User',
			passwordUpdatedAt: '2022-04-30 20:35:35',
			userCreatedAt: '2022-04-30 20:35:35',
			userName: 'test1',
			userRole: 'ADMIN',
		});

		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'User role updated to ADMIN',
			title: 'User role updated successfully',
			type: 'success',
		});

		expect(document.body).toMatchSnapshot();
	});

	it('should show error notification when role update fails', async () => {
		const notificationSpy = jest.spyOn(actualNotification, 'default');
		updateUserRole.mockRejectedValue(new Error('Role update failed'));

		let tree;
		await act(async () => {
			tree = await render(<UserRole />);
		});

		const { getByTestId, queryByTestId } = tree;

		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(getByTestId('edit-user-role')).toBeInTheDocument();
		expect(getUserRoles).toHaveBeenCalledTimes(1);

		await act(async () => {
			fireEvent.change(getByTestId('user-role-options'), {
				target: { value: 'ADMIN' },
			});
			fireEvent.click(getByTestId('update-role-btn'));
		});
		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		await act(async () => {
			fireEvent.click(getByTestId('confirmation-modal-confirm-btn'));
		});
		expect(updateUserRole).toHaveBeenCalledTimes(1);
		expect(updateUserRole).toHaveBeenCalledWith('test1', 'ADMIN');
		expect(queryByTestId('confirmation-modal')).not.toBeInTheDocument();
		expect(notificationSpy).toHaveBeenCalledTimes(1);
		expect(notificationSpy).toHaveBeenCalledWith({
			animationIn: 'animate__slideInRight',
			animationOut: 'animate__slideOutRight',
			message: 'Please try again',
			title: 'Failed to update user role',
			type: 'danger',
		});

		expect(document.body).toMatchSnapshot();
	});
});
