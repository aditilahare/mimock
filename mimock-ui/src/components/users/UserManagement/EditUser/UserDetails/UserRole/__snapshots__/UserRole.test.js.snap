// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRole should close update role confirmation modal on clicking cancel 1`] = `
<body>
  <div>
    <div
      class="flex items-center align-middle my-8 justify-between"
      data-testid="edit-user-role"
    >
      <div
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
      >
        Role
      </div>
      <div
        class="w-1/2 flex items-center align-middle mx-20"
      >
        <select
          class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
          data-testid="input-role"
        >
          <option
            data-testid="options"
            value="ADMIN"
          >
            ADMIN
          </option>
          <option
            data-testid="options"
            selected=""
            value="MANAGER"
          >
            MANAGER
          </option>
          <option
            data-testid="options"
            value="VIEWER"
          >
            VIEWER
          </option>
        </select>
        <button
          aria-label="Administrator can create and manage mocks, and also has user management privileges"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="user-role-tooltip"
          tabindex="0"
          type="button"
        >
          <svg
            class="text-gray-500 text-2xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
            />
            <path
              d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mx-4 text-gray-700 w-fit bg-green-300"
          data-testid="update-role-btn"
        >
          Update role
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`UserRole should render user role 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-role"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Role
    </div>
    <div
      class="w-1/2 flex items-center align-middle mx-20"
    >
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
        data-testid="input-role"
      >
        <option
          data-testid="options"
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          data-testid="options"
          selected=""
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          data-testid="options"
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
      <button
        aria-label="Manager can create and manage mocks, but cannot manage users"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="user-role-tooltip"
        tabindex="0"
        type="button"
      >
        <svg
          class="text-gray-500 text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
          />
          <path
            d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`UserRole should show error notification when role update fails 1`] = `
<body>
  <div>
    <div
      class="flex items-center align-middle my-8 justify-between"
      data-testid="edit-user-role"
    >
      <div
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
      >
        Role
      </div>
      <div
        class="w-1/2 flex items-center align-middle mx-20"
      >
        <select
          class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
          data-testid="input-role"
        >
          <option
            data-testid="options"
            value="ADMIN"
          >
            ADMIN
          </option>
          <option
            data-testid="options"
            selected=""
            value="MANAGER"
          >
            MANAGER
          </option>
          <option
            data-testid="options"
            value="VIEWER"
          >
            VIEWER
          </option>
        </select>
        <button
          aria-label="Administrator can create and manage mocks, and also has user management privileges"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="user-role-tooltip"
          tabindex="0"
          type="button"
        >
          <svg
            class="text-gray-500 text-2xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
            />
            <path
              d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mx-4 text-gray-700 w-fit bg-green-300"
          data-testid="update-role-btn"
        >
          Update role
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`UserRole should show update role button on changing user role 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8 justify-between"
    data-testid="edit-user-role"
  >
    <div
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
    >
      Role
    </div>
    <div
      class="w-1/2 flex items-center align-middle mx-20"
    >
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
        data-testid="input-role"
      >
        <option
          data-testid="options"
          value="ADMIN"
        >
          ADMIN
        </option>
        <option
          data-testid="options"
          selected=""
          value="MANAGER"
        >
          MANAGER
        </option>
        <option
          data-testid="options"
          value="VIEWER"
        >
          VIEWER
        </option>
      </select>
      <button
        aria-label="Administrator can create and manage mocks, and also has user management privileges"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        data-testid="user-role-tooltip"
        tabindex="0"
        type="button"
      >
        <svg
          class="text-gray-500 text-2xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
          />
          <path
            d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mx-4 text-gray-700 w-fit bg-green-300"
        data-testid="update-role-btn"
      >
        Update role
      </button>
    </div>
  </div>
</div>
`;

exports[`UserRole should show update role confirmation modal on clicking update role button 1`] = `
<body>
  <div>
    <div
      class="flex items-center align-middle my-8 justify-between"
      data-testid="edit-user-role"
    >
      <div
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
      >
        Role
      </div>
      <div
        class="w-1/2 flex items-center align-middle mx-20"
      >
        <select
          class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
          data-testid="input-role"
        >
          <option
            data-testid="options"
            value="ADMIN"
          >
            ADMIN
          </option>
          <option
            data-testid="options"
            selected=""
            value="MANAGER"
          >
            MANAGER
          </option>
          <option
            data-testid="options"
            value="VIEWER"
          >
            VIEWER
          </option>
        </select>
        <button
          aria-label="Administrator can create and manage mocks, and also has user management privileges"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="user-role-tooltip"
          tabindex="0"
          type="button"
        >
          <svg
            class="text-gray-500 text-2xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
            />
            <path
              d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mx-4 text-gray-700 w-fit bg-green-300"
          data-testid="update-role-btn"
        >
          Update role
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="xl:w-1/3 lg:w-1/3 md:w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want to update user role to "ADMIN" ?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          Cancel
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Update user role
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`UserRole should update role on clicking confirm button 1`] = `
<body>
  <div>
    <div
      class="flex items-center align-middle my-8 justify-between"
      data-testid="edit-user-role"
    >
      <div
        class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600 mx-20"
      >
        Role
      </div>
      <div
        class="w-1/2 flex items-center align-middle mx-20"
      >
        <select
          class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
          data-testid="input-role"
        >
          <option
            data-testid="options"
            value="ADMIN"
          >
            ADMIN
          </option>
          <option
            data-testid="options"
            selected=""
            value="MANAGER"
          >
            MANAGER
          </option>
          <option
            data-testid="options"
            value="VIEWER"
          >
            VIEWER
          </option>
        </select>
        <button
          aria-label="Administrator can create and manage mocks, and also has user management privileges"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="user-role-tooltip"
          tabindex="0"
          type="button"
        >
          <svg
            class="text-gray-500 text-2xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
            />
            <path
              d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="flex items-center justify-center gap-1 px-9 py-2 mx-auto shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-110 mx-4 text-gray-700 w-fit bg-green-300"
          data-testid="update-role-btn"
        >
          Update role
        </button>
      </div>
    </div>
  </div>
</body>
`;
